### PR TITLE
remove the unpooled UdpTransport

### PR DESCRIPTION
**Pull request by AnthonySteele**
_Summarise the changes this Pull Request makes._

remove the unpooled `UdpTransport` and keep the `PooledUdpTransport`
as pooled is preferred and we don't need 2 ways to do the same thing.

_Please include a reference to a GitHub issue if appropriate._

Follow on from #113

### CODE REVIEW

File: src/JustEat.StatsD/Transport.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+namespace JustEat.StatsD
+{
+    internal static class Transport
```
Comment (martincostello) [inline src/JustEat.StatsD/Transport.cs]: `SocketFactory`?

File: src/JustEat.StatsD/Transport.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        internal static Socket UdpSocket()
```
Comment (martincostello) [inline src/JustEat.StatsD/Transport.cs]: `ForUdp()`?

File: src/JustEat.StatsD/Transport.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                socket.SendBufferSize = 0;
+            }
+#else
+            socket.SendBufferSize = 0;
+#endif
+
+            return socket;
+        }
+
+        internal static Socket IpSocket()
```
Comment (martincostello) [inline src/JustEat.StatsD/Transport.cs]: `ForIP()`?

**Comment by martincostello** [comment]
Suggest in a PR after this one to rename `PooledUdpTransport` to `UdpTransport` - the pooling is now just an implementation detail, so I don't think it needs to be in the type name.

File: src/JustEat.StatsD/Transport.cs (hunk 1 above)
Comment (AnthonySteele) [reply src/JustEat.StatsD/Transport.cs]: I have changed all 3 as per comments

**Pull request merged**